fairscale/optim/adascale.py

```diff
@@ -38,27 +38,36 @@
 import torch
 from torch.autograd import Variable
 import torch.distributed as dist
+from torch.optim import Optimizer
 
 
-class AdaScale(object):
+class AdaScale(Optimizer):
     """
     Implements the AdaScale_ algorithm for scaling the learning rate for
     distributed and large batch size training. Can be used in combination with
     ``torch.nn.parallel.DistributedDataParallel`` and ``torch.optim.SGD``.
 
+    Subclass `Optimizer` so that `torch.optim.lr_scheduler` can work.
+
     .. _AdaScale: https://proceedings.icml.cc/static/paper_files/icml/2020/4682-Supplemental.pdf
 
+    Here is an overly simple example. For `torch.optim.lr_scheduler`
+    based example, please checkout our tutorial.
+
     .. code-block:: python
 
         optim = torch.optim.SGD(model.parameters(), lr=0.001)
         model = DistributedDataParallel(model)
         adascale = AdaScale(optim)
 
+        step = 0
         for epoch in ...:
             for batch in ...:
                 optim.zero_grad()
                 loss = ...
                 loss.backward()
+                step += adascale.gain()
+                update_lr(step)
```

Comment:
Would it make sense to use the same example code across Readme, rst, and pydoc?

Reply:
I was thinking that with a varieties of examples, maybe it helps the user more in different situations?

Reply:
Sure, makes sense, though it could also potentially cause confusion. I'd suggest if we want to show multiple examples, to group them together in the same place (example 1, example 2, ...) and clearly point out that there's more than one way to use this.

Reply:
Good idea. How about the following?

* we keep the README.md example more "official" (the lr_schedule one)
* keep the tutorial example as is
* here, we have two examples, one for each style

Reply:
Sounds great!